src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -706,7 +719,10 @@ protected function redirectAfterSave($isNewRecord)
         } elseif ($this->gridField->getList()->byID($this->record->ID)) {
             // Return new view, as we can't do a "virtual redirect" via the CMS Ajax
             // to the same URL (it assumes that its content is already current, and doesn't reload)
-            return $this->edit($controller->getRequest());
+            $message = $controller->getRequest()->getHeader('X-Status') ?? rawurlencode(_t(__CLASS__ . '.SAVEDUP', 'Saved successfully') ?? '');
+            $response = new HTTPResponse($this->edit($controller->getRequest()));
+            $response->addHeader('X-Status', $message);
+            return $response;
```

Comment:
```suggestion
            $controller->getResponse()->addHeader('X-Status', $message);
            return $this->edit($controller->getRequest());
```
We can't return a new response here, because the controller that ultimately returns the response could have already made some modifications to its response in its init method, for example, which this would ignore. We have to add headers to the existing response.

With this, it might be worth double checking if you still need the change to `LeftAndMain` in the admin PR.

Reply:
Updated

Reply:
Updated